app/components/Snackbar/index.js

```diff
@@ -30,17 +30,23 @@ const snackbarClasses = ({ type }) => ({
 
 @autobind
 class Snackbar extends React.Component {
-
   constructor(props) {
     super(props);
     this.hideTimer = null;
     this.state = {
       message: props.messages.length > 0
         ? props.messages[props.messages.length-1]
-        : null
+        : null,
+      delay: 250, // delay in ms to render component
+      isWaiting: true,
     };
   }
 
+  componentDidMount() {
+    setTimeout(() => {
+      this.setState({isWaiting: false}); }, this.state.delay);
+  }
+
   componentWillReceiveProps(nextProps) {
```

Comment:
You need to add the delay here (a bit below actually, after the component has determined this is a new message). 

You need to store the new message in an intermediate attribute of the class (*not* the state yet, so we don't trigger a render) and start the 250ms timer here (if it's not already running - this is important). Once the timer callback is called, then you update the state with the new message (which will trigger the render).

If there are multiple messages being received within the 250ms window (which you'll detect if `componentWillReceiveProps` gets called while the timer is set), your options are:

1. Discard the new messages and display the first one (and keep the timer running)
2. Discard the old one and display the most recent (and keep the timer running)
3. Discard the old one and display the most recent (but restart the timer)